runtime/doc/vim9class.txt

```diff
@@ -972,8 +972,8 @@ The following example shows an enum with object variables and methods: >
     echo Planet.Earth.has_rings
 <
 						*E1421* *E1423* *E1424* *E1425*
-Enums and their values are immutable. They cannot be modified after
-declaration and cannot be utilized as numerical or string types.
+Enums and their values are immutable; however, mutable components are allowed
+for enum values.  They cannot be utilized as numerical or string types.
```

Comment:
I don't understand that sentence. First we are saying *... their values are immutable*  and just a few words further we are saying that *mutable components are allowed for enum values*? What does that mean and what exactly mutable for enum values?

Reply:
Sigh... I wish I could express it better.

Well, given these types:

```
vim9script

class Class
    var text: string = "Class"
endclass

enum Enum 
    INSTANCE

    var value: number = 0

    def new()
    enddef

    def SetValue(value: number)
	this.value = value
    enddef
endenum
```

> Enums and their values are immutable

Neither enumerations are extendable by a class (and so no  
new state can be acquired through class inheritance),

```
# E1416
enum Enum extends Class
    INSTANCE
endenum
```

nor any _new_ instances beyond those already declared can be  
created.

```
# E1325
Enum.new()
```

But the state of any enumeration value can be changed when  
the said state is mutable.

```
Enum.INSTANCE.SetValue(5)
echo Enum.INSTANCE
```

Reply:
I've rewritten vague parts.  Hopefully, the changed text is
more clear now.